pkg/tools/otelcol-config/timezone.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/mikefarah/yq/v4/pkg/yqlib"
+	"gopkg.in/yaml.v3"
+)
+
+func SetTimezoneAction(ctx *actionContext) error {
+	conf, err := ReadConfigDir(ctx.ConfigDir)
+	if err != nil {
+		return err
+	}
+
+	var writer func([]byte) (int, error)
+	var config []byte
+
+	switch {
+	case conf.SumologicRemote != nil || ctx.Flags.EnableRemoteControl:
+		writer = ctx.WriteSumologicRemote
+		config = conf.SumologicRemote
+	case ctx.Flags.Override:
+		writer = ctx.WriteConfDOverrides
+		config = conf.ConfD[ConfDOverrides]
+	default:
+		writer = ctx.WriteConfD
+		config = conf.ConfD[ConfDSettings]
+	}
+
+	encoder := yqlib.YamlFormat.EncoderFactory()
+	decoder := yqlib.YamlFormat.DecoderFactory()
+	eval := yqlib.NewStringEvaluator()
+
+	if len(config) == 0 {
```

Comment:
It would be nice to add a test to cover the expected behavior when this config file isn't present, example - https://github.com/SumoLogic/sumologic-otel-collector/blob/c96bd9fc8eaf3a215d3f2cf50d5d05ee88c36a42/pkg/tools/otelcol-config/remote_control_test.go#L75

This would hopefully test the default timezone scenario.

Also, show we include the timezone in the initial config here? - https://github.com/SumoLogic/sumologic-otel-collector/blob/c96bd9fc8eaf3a215d3f2cf50d5d05ee88c36a42/pkg/tools/otelcol-config/remote_control.go#L69

cc: @echlebek

Reply:
it is okay not to include default timezone since API also have default "UTC"

Reply:
Added more test

Reply:
IMO it makes sense for the collector to have a default timezone written to its config when none is set and this should be documented in the readme. The sumo extension doesn't seem to set a default timezone either.

Reply:
Added default also